besu-plugins/linea-sequencer/sequencer/src/main/java/net/consensys/linea/sequencer/txselection/selectors/TransactionEventFilter.java

```diff
@@ -0,0 +1,26 @@
+package net.consensys.linea.sequencer.txselection.selectors;
+
+import org.hyperledger.besu.datatypes.Address;
+import org.hyperledger.besu.evm.log.Log;
+import org.hyperledger.besu.evm.log.LogTopic;
+
+public record TransactionEventFilter(
+    Address contractAddress, LogTopic topic0, LogTopic topic1, LogTopic topic2, LogTopic topic3) {
+  /**
+   * Checks whether the supplied log matches this TransactionEventFilter
+   *
+   * @param log the Log to be checked
+   * @return true if the supplied log matches this TransactionEventFilter, false otherwise
+   */
+  public boolean matches(final Log log) {
+    return log.getLogger().equals(contractAddress)
+        && (topic0 == null
+            || (log.getTopics().size() >= 1 && log.getTopics().get(0).equals(topic0)))
+        && (topic1 == null
+            || (log.getTopics().size() >= 2 && log.getTopics().get(1).equals(topic1)))
+        && (topic2 == null
+            || (log.getTopics().size() >= 3 && log.getTopics().get(2).equals(topic2)))
+        && (topic3 == null
+            || (log.getTopics().size() >= 4 && log.getTopics().get(3).equals(topic3)));
+  }
```

Comment:
there is an edge case, that I am not sure is intended or not, basically if the event filter only has the contract set and all the log topics are null, the `matches` will always return true for that contract.
Should we require that at least one log topic is configured for a contract? 
Otherwise I think we are basically denying any interaction with that contract, that could also be a feature, just want to clarify the requirement